File: invesdwin-scripting-parent/invesdwin-scripting-julia/invesdwin-scripting-julia-runtime-contract/src/test/java/de/invesdwin/scripting/julia/runtime/contract/InputsAndResultsTestBoolean.jl
println("getBoolean")
if isdefined(Main, :getBoolean) && !isnothing(getBoolean)
	error("getBoolean already defined!")
end
getBoolean = putBoolean
println(typeof(getBoolean))
println(getBoolean)
if typeof(getBoolean) != Bool
	error("getBoolean not Bool!")
end

println("getBooleanVector")
if isdefined(Main, :getBooleanVector) && !isnothing(getBooleanVector)
	error("getBooleanVector already defined!")
end
getBooleanVector = putBooleanVector
println(eltype(getBooleanVector))
println(getBooleanVector)
if eltype(getBooleanVector) != Bool 
	error("getBooleanVector not Bool!")
end

println("getBooleanVectorAsList")
if isdefined(Main, :getBooleanVectorAsList) && !isnothing(getBooleanVectorAsList)
	error("getBooleanVectorAsList already defined!")
end
getBooleanVectorAsList = putBooleanVectorAsList
println(eltype(getBooleanVectorAsList))
println(getBooleanVectorAsList)
if eltype(getBooleanVectorAsList) != Bool
	error("getBooleanVectorAsList not Bool!")
end

println("getBooleanMatrix")
if isdefined(Main, :getBooleanMatrix) && !isnothing(getBooleanMatrix)
	error("getBooleanMatrix already defined!")
end
getBooleanMatrix = putBooleanMatrix
println(eltype(getBooleanMatrix))
println(getBooleanMatrix)
if eltype(getBooleanMatrix) != Bool
	error("getBooleanMatrix not Bool!")
end

println("getBooleanMatrixAsList")
if isdefined(Main, :getBooleanMatrixAsList) && !isnothing(getBooleanMatrixAsList)
	error("getBooleanMatrixAsList already defined!")
end
getBooleanMatrixAsList = putBooleanMatrixAsList
println(eltype(getBooleanMatrixAsList))
println(getBooleanMatrixAsList)
if eltype(getBooleanMatrixAsList) != Bool
	error("getBooleanMatrixAsList not Bool!")
end

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/InputsAndResultsTestBoolean.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getBoolean")
if(getBoolean ~= nil) then
	error("getBoolean already defined!")
end
getBoolean = putBoolean
getBooleanType = type(getBoolean)
print(getBooleanType)
print(getBoolean)
if(getBooleanType ~= "boolean") then
	error("getBoolean not Boolean!")
end

print("getBooleanVector")
if(type(getBooleanVector) ~= "nil") then
	error("getBooleanVector already defined!")
end
getBooleanVector = putBooleanVector
getBooleanVectorType = type(getBooleanVector[1])
print(getBooleanVectorType)
print(arraysToString(getBooleanVector))
if(getBooleanVectorType ~= "boolean") then
	error("getBooleanVector not Boolean!")
end

print("getBooleanVectorAsList")
if(type(getBooleanVectorAsList) ~= "nil") then
	error("getBooleanVectorAsList already defined!")
end
getBooleanVectorAsList = putBooleanVectorAsList
getBooleanVectorAsListType = type(getBooleanVectorAsList[1])
print(getBooleanVectorAsListType)
print(arraysToString(getBooleanVectorAsList))
if(getBooleanVectorAsListType ~= "boolean") then
	error("getBooleanVectorAsList not Boolean!")
end

print("getBooleanMatrix")
if(type(getBooleanMatrix) ~= "nil") then
	error("getBooleanMatrix already defined!")
end
getBooleanMatrix = putBooleanMatrix
getBooleanMatrixType = type(getBooleanMatrix[1][1])
print(getBooleanMatrixType)
print(arraysToString(getBooleanMatrix))
if(getBooleanMatrixType ~= "boolean") then
	error("getBooleanMatrix not Boolean!")
end

print("getBooleanMatrixAsList")
if(type(getBooleanMatrixAsList) ~= "nil") then
	error("getBooleanMatrixAsList already defined!")
end
getBooleanMatrixAsList = putBooleanMatrixAsList
getBooleanMatrixAsListType = type(getBooleanMatrixAsList[1][1])
print(getBooleanMatrixAsListType)
print(arraysToString(getBooleanMatrixAsList))
if(getBooleanMatrixAsListType ~= "boolean") then
	error("getBooleanMatrixAsList not Boolean!")
end
